frb_codegen/src/generator/dart/ty_struct.rs

```diff
@@ -80,9 +80,9 @@ impl TypeDartGeneratorTrait for TypeStructRefGenerator<'_> {
 
         format!(
             "final arr = raw as List<dynamic>;
-                if (arr.length != {}) throw Exception('unexpected arr length: expect {} but see ${{arr.length}}');
+                if (arr.{}) throw Exception('unexpected arr length: expect {} but see ${{arr.length}}');
                 return {}({});",
-            s.fields.len(),
+            if s.is_empty { "isNotEmpty".to_string()} else { format!("length != {}", s.fields.len()) },
```

Comment:
uber-nit: The original, `arr.length != ...` also LGTM, since it is quite uniform :) If you are worried about linting errors, just add one word into `// ignore_for_file: ...` at the *top* of the generated bridge_generated.io.dart etc.